### DISCUSSION TITLE
Why "colour.xy_to_CCT(xy, 'McCamy 1992')" definition return value doesn't match the formula?

### DISCUSSION THREAD

**Discussion by lishichengyan** [colour-science/colour#821] [Q&A] [answered]
Hi there, thanks for this wonderful package! I have a question regarding the ```colour.xy_to_CCT(xy, 'McCamy 1992')``` function. I'm using ```colour.xy_to_CCT(xy, 'McCamy 1992')``` to calculate the CCT given a RGB vector and here is the code:

```
import numpy as np
import colour
RGB = np.array([231.0, 260.0, 95.0])
XYZ = colour.sRGB_to_XYZ(RGB)
xy = colour.XYZ_to_xy(XYZ)
CCT = colour.xy_to_CCT(xy, 'McCamy 1992') # Returns 4478.34619011
```

However according to McCamy's equations:

```
CCT= 449*n^3+ 3525*n^2+ 6823.3*n+ 5520.33, where
n = (x − 0.3320) / (0.1858 − y) 
x = X/(X+Y+Z)
y = Y/(X+Y+Z)
X = (-0.14282)(R) + (1.54924)(G) + (-0.95641)(B) 
Y = (-0.32466)(R) + (1.57837)(G) + (-0.73191)(B) = Illuminance
Z = (-0.68202)(R) + (0.77073)(G) + (0.56332)(B) 
```

The CCT for [231.0, 260.0, 95.0] should be 3121K, as described in this paper (https://ams.com/documents/20143/80162/TCS34xx_AN000517_1-00.pdf/1efe49f7-4f92-ba88-ca7c-5121691daff7).

I'm wondering if there's something wrong with my understanding. The version of my colour-science is 0.3.16. Thanks for your attention!

**Comment by thomasmansencal** [comment, marked as answer]
Hi @lishichengyan,

So your code is not quite correct for two reasons:

1. The `colour.sRGB_to_XYZ` convenience definition assumes that data is encoded non-linearly and thus will decode it. You should pass `apply_cctf_decoding=False`.
2. The previous point is actually irrelevant because the RGB space from the paper is from a sensor, not sRGB thus you need to use the given 3x3 RGB to XYZ matrix.

Keeping all that in mind:

```python
import colour
import numpy as np

RGB = np.array([231.0, 260.0, 95.0])
M = [
    [-0.14282, +1.54924, -0.95641],
    [-0.32466, +1.57837, -0.73191],
    [-0.68202, +0.77073, +0.56332 ],
]
XYZ = colour.algebra.vector_dot(M, RGB)
xy = colour.XYZ_to_xy(XYZ)
colour.xy_to_CCT(xy, 'McCamy 1992')
3121.062453413394
```

**Reply by lishichengyan** [reply]
Got it, thank you for your quick reply!
